イベント設定ページに開始・終了日付、募集チーム数、目的・その他入力欄を追加

toshinori-m opened the pull request:
イベント設定ページの実装第2段階として、以下の項目を追加しました。

### 変更内容
- 以下がイベント全項目
  - 競技種別（単一選択）
  - 種目（複数選択）
  - 都道府県（単一選択）
  - 対象年齢（複数選択）
  - イベント名（テキスト入力）
  - イベントURL（URL入力）
  - 地域（テキスト入力）
  - 性別（ラジオボタン）
  - 開始日付（date） ←（本PRにて追加）
  - 終了日付（date） ←（本PRにて追加）
  - 募集チーム数（数値入力） ←（本PRにて追加）
  - イベント目的（テキスト入力） ←（本PRにて追加）
  - その他（テキスト入力） ←（本PRにて追加）
- エラー時のメッセージ表示を実装
- 文字数制限に応じた警告表示を実装（イベント名、地域）
- 入力内容をまとめて送信する `handleSubmit` 関数を追加

### 備考
- api送信処理の実装は下のPRにて対応予定です。
  - [イベント設定ページにapi送信処理を実装](https://github.com/toshinori-m/stay_connect/issues/200)

close https://github.com/toshinori-m/stay_connect/issues/193

toshinori-m committed:
feat: イベント設定ページに日付・チーム数・目的・その他入力欄と送信関数を追加

hirosi1900day commented on frontend-react/src/components/ui/InputField.tsx:
```diff
@@ -2,12 +2,13 @@ import { ReactNode } from "react"
 interface InputFieldProps {
   label: ReactNode
-  type?: "text" | "email" | "password" | "url" | "radio"
+  type?: "text" | "email" | "password" | "url" | "radio" | "date" | "number" | "text"
```
nits
reactからHTMLInputTypeAttributeをimportしたりする方がシンプルだったり？？
型アサーションが必要になるかもしれないのでそこはデメリットではあるので悩ましい

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/5e49227a673b2775d584fdb655e701a17c92093d/types/react/index.d.ts#L3072C10-L3072C32

hirosi1900day commented on frontend-react/src/pages/eventPages/EventSettingPage.tsx:
```diff
@@ -73,132 +78,215 @@ export default function EventSettingPage() {
       <div className="w-full md:w-3/5 xl:w-2/5 shadow-gray-200 bg-sky-100 rounded-lg">
         <h2 className="text-center mb-10 pt-10 font-bold text-3xl text-blue-600">イベント設定</h2>
-        <form className="px-4 md:px-0 text-center">
+        <form className="px-4 md:px-0 text-center" onSubmit={handleSubmit}>
```
nits
まさにここがuseStateActionの使い場所ながしますね

hirosi1900day commented on frontend-react/src/pages/eventPages/EventSettingPage.tsx:
```diff
@@ -223,9 +311,77 @@ export default function EventSettingPage() {
+          {/* 登録ボタン */}
+          <div className="text-center mb-5">
+            <button className="btn-ok my-4 md:mb-0 md:mr-4">登録する</button>
+          </div>
```
buttonは共通コンポーネントとしてプロジェクトの中である程度統一したボタンデザインにした方がよかったりしませんかね？？

toshinori-m replied:
共通コンポーネントとして作成したButton.tsxの他の箇所の使用の修正は下のissueで行います。
- [SearchForm.tsxのスタイル統一（共通セレクトコンポーネント利用、jsx内のロジックを無くす、Button.tsx利用など）](https://github.com/toshinori-m/stay_connect/issues/201)

toshinori-m committed:
refactor: InputFieldの型を統一、EventSettingPageのUIロジックを整理

- InputFieldでHTMLInputTypeAttributeを使用するよう修正
- EventSettingPageにuseStateActionを適用
- EventSettingPageで共通Buttonコンポーネントを使用

hirosi1900day commented on frontend-react/src/components/ui/Button.tsx:
```diff
@@ -0,0 +1,20 @@
+    signup: "btn-signup",
+    ok: "btn-ok",
+    login: "btn-login",
+  }[variant]
```
[nits]
共通デザインなのでsignupようなどのように細かい用途でデザインを切り替えるというよりはsubmit用 とか見たいな感じでおおなくくりでデザインを切り替えるようにした方が良さそうな気がしますね(ボタンにアプリの仕様など関心を持たせない方がデザインが統一化できて良さそうといったイメージ)

ボタンの丸みとかは統一しつつ色が2~3種類くらいに分かれているだけみたいな方がシンプルな気がしてます

hirosi1900day commented on frontend-react/src/hooks/useEventSubmit.ts:
hookにしては処理の関心がEventSettingPage.tsxによっているように見えますが、これは他のところでも再利用などをする想定でしょうか？？

再利用の予定がないのであれば元々の実装をベースにhandleSubmitを定義してそれをuseStateActionの引数に指定する形にしてEventSettingPage.tsxに処理を書いた方が良いかなと思ったりしました
(特定のpageに関心がより過ぎたものをわざわざhook配下に切り出すのは個人的にはあんまりメリットを感じることができなかったため指摘してみました。再利用できるような処理であればここに配置するのは賛成です)

toshinori-m committed:
refactor: ボタンスタイルの統一とクラス名の再構成（variant/size対応に変更）

toshinori-m committed:
refactor: useActionStateをEventSettingPage.tsxに統合しバリデーションと送信処理を整理

hirosi1900day commented on frontend-react/src/pages/eventPages/EventSettingPage.tsx:
```diff
@@ -212,6 +278,7 @@ export default function EventSettingPage() {
                 ].map((option) => (
                   <div key={option.value}>
                     <input
```
ここをInputFieldを使わなかった理由はありますか？？

hirosi1900day commented on frontend-react/src/components/ui/InputField.tsx:
```diff
@@ -1,34 +1,40 @@
 }
 
-export default function InputField({ 
+export default function InputField({
```
radioボタン等はデザインやhtmlの構成が違うのでそう言ったものと区別がつく命名にしてradio系のinputは別のコンポーネントに切り出した方が良さそう

hirosi1900day reviewed (approved):
良さそう

hirosi1900day commented:
approveは間違えた

toshinori-m committed:
Refactor: ラジオボタンの構造を整理し、RadioGroupField コンポーネントとして分離

toshinori-m committed:
Fix: フォームのバリデーションエラー後に「競技名」と「都道府県」が正しくリセットされない問題を修正

- useActionStateの返却データにformDataを追加し、エラー時も現在のフォーム状態を維持できるように変更
- フォーム送信後にuseEffectでformStateを更新する処理を追加
- 選択状態を明示的に初期化することで、バリデーションエラー時に誤って選択されたように見える問題を解消

toshinori-m commented:
### 修正内容について
フォームバリデーションエラー発生時に「競技名」や「都道府県」が未選択にも関わらず、選択されたように見えてしまう問題を修正しています。
### 問題の概要
- `useActionState` + `form action={...}` を使用している場合、フォーム送信後にバリデーションエラーが発生すると、Reactの再レンダリングによって `formState` が正しく維持されず、セレクトボックスが初期データ（例: `initialData[0]`）を選択しているように見えてしまう挙動が発生していました。
### 原因の考察
- `useActionState` を使用した場合、サーバーアクション後に React がフォームコンポーネントを再描画します。
- その際、`formState` を明示的に保持しないと、状態が初期化され、セレクトボックスが初期値を自動的に選択したように見える原因となっていました。
### 解決策
- サーバーアクション内で現在のフォーム状態（`formState`）を `formData` として返却し、バリデーションエラーがある場合に状態を保持。
- `useEffect` を使って `actionState.formData` が更新された際に `formState` を復元するようにして、再描画後も選択状態が維持されるようにしました。

よろしくお願いします！

toshinori-m committed:
refactor: 関数呼び出しの順序を変更し、処理の整合性を改善

toshinori-m commented:
@hirosi1900day 
お忙しいところ恐縮ですが、こちらのPRのレビューをお願いできますでしょうか？
ボタンの共通化などをこちらのprで行っています

よろしくお願いします
